hal/src/rtl872x/exflash_hal.cpp

```diff
@@ -291,36 +595,61 @@ static bool is_block_erased(uintptr_t addr, size_t size) {
 }
 
 __attribute__((section(".ram.text"), noinline))
-static int erase_common(uintptr_t start_addr, size_t num_blocks, int len) {
-    ExFlashLock lk;
-
-    const size_t block_length = len == EraseSector ? 4096 : 64 * 1024;
-
-    /* Round down to the nearest block */
-    start_addr = ((start_addr / block_length) * block_length);
-
-    for (size_t i = 0; i < num_blocks; i++) {
-        if (!is_block_erased(start_addr + i * block_length, block_length)) {
-            FLASH_Erase(len, start_addr + i * block_length);
-        }
+static int erase_common(uintptr_t address, hal_exflash_cmd_t cmd) {
+    ExFlashLock lk(false);
+    size_t size = 0;
+    if (cmd == GENERIC_FLASH_CMD_SE) {
+        size = 4 * 1024;
+    } else if (cmd == GENERIC_FLASH_CMD_BE32K) {
+        size = 32 * 1024;
+    } else if (cmd == GENERIC_FLASH_CMD_BE64K) {
+        size = 64 * 1024;
+    } else {
+        return SYSTEM_ERROR_INVALID_ARGUMENT;
     }
+    address = ((address / (size)) * (size));
+    if (is_block_erased(address, (size))) {
+        return size;
+    }
+    RtkSpicFlashConfigGuard guard;
 
-    // LOG_DEBUG(ERROR, "Erased %lu %lukB blocks starting from %" PRIxPTR,
-    //           num_blocks, block_length / 1024, start_addr);
-
-    dcacheCleanInvalidateAligned(SPI_FLASH_BASE + start_addr, block_length * num_blocks);
-
-    return SYSTEM_ERROR_NONE;
+    rtkSpiFlashSendCmd(GENERIC_FLASH_CMD_WREN);
+    /* Disable SPI_FLASH */
+    SPIC->ssienr = 0;
+    rtkSpiSetTxMode();
```

Comment:
This means that there is no data on the other IO lines? IE this is just a single line command + address transmission?

Reply:
Yup
![image](https://github.com/user-attachments/assets/abd5993e-bffc-494e-8161-e7165b574fb0)